src/prefect/serialization/result.py

```diff
@@ -1,16 +1,50 @@
 from marshmallow import fields
 
-from prefect.engine import result
+from prefect.engine import result, results
 from prefect.serialization.result_handlers import ResultHandlerSchema
 from prefect.utilities.serialization import JSONCompatible, ObjectSchema, OneOfSchema
 
 
+class GCSResultSchema(ObjectSchema):
+    class Meta:
+        object_class = results.GCSResult
+
+    filepath = fields.String(allow_none=False)
+    bucket = fields.String(allow_none=False)
+    credentials_secret = fields.String(allow_none=True)
+
+
+class PrefectResultSchema(ObjectSchema):
+    class Meta:
+        object_class = results.PrefectResult
+
+    value = fields.String(allow_none=True)
+
+
+class S3ResultSchema(ObjectSchema):
+    class Meta:
+        object_class = results.S3Result
+
+    filepath = fields.String(allow_none=False)
+    bucket = fields.String(allow_none=False)
+    credentials_secret = fields.String(allow_none=True)
+
+
+class ConstantResultSchema(ObjectSchema):
+    class Meta:
+        object_class = results.ConstantResult
+
+
+class ResultSchema(ObjectSchema):
```

Comment:
I've been slowly changing most every reference to instantiating a base class `Result` in the pipeline to instantiating a `PrefectResult` in the pipeline. Do we really need serializers for base class Results? I empirically observed and discussed on a call with @cicdw right now this is necessary for as long as `Result` is the default for States:  https://github.com/PrefectHQ/prefect/blob/d593e0e7a0a02d25f441b3d0e942614471ee604a/src/prefect/engine/state.py#L85-L90
Do we think we should change that default to `PrefectResult` now?